aw_qt/manager.py

```diff
@@ -1,68 +1,128 @@
+    return None
+
 
-def _locate_executable(name: str) -> List[str]:
+def _is_system_module(name: str) -> bool:
+    """Checks if a module with a particular name exists in PATH"""
+    return shutil.which(name) is not None
+
+
+def _locate_executable(name: str) -> Optional[str]:
     """
-    Will start module from localdir if present there,
-    otherwise will try to call what is available in PATH.
+    Will return the path to the executable if bundled,
+    otherwise returns the name if it is available in PATH.
 
-    Returns it as a Popen cmd list.
+    Used when calling Popen.
     """
-    curr_filepath = os.path.realpath(__file__)
-    curr_dir = os.path.dirname(curr_filepath)
-    program_dir = os.path.abspath(os.path.join(curr_dir, os.pardir))
-    search_paths = [curr_dir, program_dir, os.path.join(program_dir, name)]
+    exec_path = _locate_bundled_executable(name)
+    if exec_path is not None:  # Check if it exists in bundle
+        return exec_path
+    elif _is_system_module(name):  # Check if it's in PATH
+        return name
+    else:
+        logger.warning("Could not find module '{}' in installation directory or PATH".format(name))
+        return None
+
+
+""" Look for modules in given directory path and recursively in subdirs matching aw-* """
+def _discover_modules_in_directory(modules: List[str], search_path: str) -> None:
+    matches = glob(os.path.join(search_path, "aw-*"))
+    for match in matches:
+        if os.path.isfile(match) and os.access(match, os.X_OK):
+            name = os.path.basename(match)
+            modules.append(name)
+        elif os.path.isdir(match) and os.access(match, os.X_OK):
+            _discover_modules_in_directory(modules, match)
+        else:
+            logger.warning("Found matching file but was not executable: {}".format(match))
 
-    exec_end = ".exe" if platform.system() == "Windows" else ""
-    exec_paths = [os.path.join(path, name + exec_end) for path in search_paths]
+""" Use _discover_modules_in_directory to find all bundled modules """
+def _discover_modules_bundled() -> List[str]:
+    modules: List[str] = []
+    cwd = os.getcwd()
+    _discover_modules_in_directory(modules, cwd)
 
-    for exec_path in exec_paths:
-        if os.path.isfile(exec_path):
-            # logger.debug("Found executable for {} in: {}".format(name, exec_path))
-            return [exec_path]
-            break  # this break is redundant, but kept due to for-else semantics
+    if len(modules) > 0:
+        logger.info("Found bundled modules: {}".format(set(modules)))
     else:
-        # TODO: Actually check if it is in PATH
-        logger.debug("Trying to start {} using PATH (executable not found in: {})"
-                     .format(name, exec_paths))
-        return [name]
+        logger.info("Found no bundles modules")
+    return modules
+
+""" Find all aw- modules in PATH """
+def _discover_modules_system() -> List[str]:
+    search_paths = os.environ["PATH"].split(":")
```

Comment:
Hi, I haven't tested these changes myself but I think this line won't work on Windows, as Windows uses semicolon (`;`) to separate PATH items. Python has `os.pathsep` which returns the correct separator, so you should be able to do:
```
search_paths = os.environ["PATH"].split(os.pathsep)
```

Reply:
Actually, [get_exec_path()](https://docs.python.org/3.7/library/os.html#os.get_exec_path) might be a cleaner solution (for Python 3.2+):
```
search_paths = os.get_exec_path()
```

Credits to @martincizek.

Reply:
@RoBobik great points, needs fixing before merge.

Reply:
Yeah good points. Changed to using os.get_exec_path() in 68314f0